tests/test_open_buildings.py

```diff
@@ -1,33 +1,120 @@
 
-    def setUp(self):
-        """Set up test fixtures, if any."""
+def test_geojson_to_quadkey(aoi: Dict[str, Any]):
+    """ Tests geojson_to_quadkey() using a pre-established true value. """
+    assert geojson_to_quadkey(aoi) == '301001330310'
 
-    def tearDown(self):
-        """Tear down test fixtures, if any."""
+def test_quadkey_to_geojson():
+    """ Tests quadkey_to_geojson() using a pre-established true value. """
+    assert quadkey_to_geojson('031313131112') == {'type': 'Feature', 'geometry': {'type': 'Polygon', 'coordinates': [[[-0.17578125, 51.50874245880333], [-0.087890625, 51.50874245880333], [-0.087890625, 51.56341232867588], [-0.17578125, 51.56341232867588], [-0.17578125, 51.50874245880333]]]}}
 
-    def test_000_something(self):
-        """Test something."""
+@pytest.mark.integration
+@pytest.mark.flaky(reruns=NUM_RERUNS)
+@pytest.mark.parametrize("source", [s for s in Source])
+def test_download(source: Source, aoi: Dict[str, Any], tmp_path: Path):
+    """ Tests that the download function successfully downloads a GeoJSON file from all sources (parametrised test) into a temporary directory (teardown after test). """
+    output_file = tmp_path.joinpath(f"output_{source.name}.json")
+    download(aoi, source=source, dst=output_file, country_iso="SC")
+    assert os.path.exists(output_file)
+    assert os.path.getsize(output_file) != 0
 
-    # def test_command_line_interface(self):
-    #     """Test the CLI."""
-    #     runner = CliRunner()
-    #     result = runner.invoke(cli.main)
-    #     assert result.exit_code == 0
-    #     assert 'open_buildings.cli.main' in result.output
-    #     help_result = runner.invoke(cli.main, ['--help'])
-    #     assert help_result.exit_code == 0
-    #     assert '--help  Show this message and exit.' in help_result.output
+@pytest.mark.integration
+@pytest.mark.flaky(reruns=NUM_RERUNS)
+def test_download_no_output(aoi: Dict[str, Any], tmp_path: Path):
+    """ Test that no empty output file gets created if a query doesn't return anything (in this case because a wrong country_iso argument is given.) """
+    output_file = tmp_path.joinpath("no_output.json")
+    download(aoi, dst=output_file, country_iso="AI") # wrong country, aoi is in SC, not Anguilla
+    assert not os.path.exists(output_file)
+
+@pytest.mark.integration
+@pytest.mark.flaky(reruns=NUM_RERUNS)
+def test_download_directory(aoi: Dict[str, Any], tmp_path: Path):
+    """ Test that, if a directory is passed, the output gets downloaded to a default file name in that directory. """
+    download(aoi, dst=tmp_path, country_iso="SC")
+    assert os.path.exists(tmp_path.joinpath("buildings.json"))
+    assert os.path.getsize(tmp_path.joinpath("buildings.json")) != 0
+
+@pytest.mark.integration
+@pytest.mark.flaky(reruns=NUM_RERUNS)
+@pytest.mark.parametrize("format", [f for f in Format if format != Format.SHAPEFILE]) # fails for shapefile!
```

Comment:
We're getting: 
```
[gw5] linux -- Python 3.10.13 /opt/hostedtoolcache/Python/3.10.13/x64/bin/python3
worker 'gw5' crashed while running 'tests/test_open_buildings.py::test_download_format[Format.SHAPEFILE]'
=========================== short test summary info ============================
FAILED tests/test_open_buildings.py::test_download_format[Format.SHAPEFILE]
```
I was going to go into the code to skip testing shapefile, but it seems to me the line above should do that, so I'm not sure what to do other than remove shapefile as an output format, but that seems silly since it works on mac and windows.